cmd/infra/azure/create.go

```diff
@@ -431,17 +445,60 @@ func createVirtualNetwork(ctx context.Context, subscriptionID string, resourceGr
 		},
 	}, nil)
 	if err != nil {
-		return "", "", "", fmt.Errorf("failed to create vnet: %w", err)
+		return armnetwork.VirtualNetworksClientCreateOrUpdateResponse{}, fmt.Errorf("failed to create vnet: %w", err)
 	}
 	vnet, err := vnetFuture.PollUntilDone(ctx, nil)
 	if err != nil {
-		return "", "", "", fmt.Errorf("failed to wait for vnet creation: %w", err)
+		return armnetwork.VirtualNetworksClientCreateOrUpdateResponse{}, fmt.Errorf("failed to wait for vnet creation: %w", err)
+	}
+
+	if vnet.ID == nil || vnet.Name == nil {
```

Comment:
these checks are similar to the ones from getClientsVirtualNetwork, do you think its worth creating a function to do these checks so we can reuse it?

Reply:
Hmm looked at that but the issue is `vnet` is a different Azure type for each function